type-strings-into-constants Extracted blurr types into string constants.

vigbk opened the pull request:
**PR Checklist**

_Please make sure to review and check all of these items:_

- [x] Tests for the changes have been added
- [x] Docs have been added / updated

**What kind of change does this PR introduce?**

_Describe the big picture of your changes here_

**Other information**

fixes #XYZ

vigbk commented:
I think we should have some consistent naming in our types. Sometimes we append the type to the end sometimes we don't. eg: Blurr:Transform:Streaming, Blurr:Aggregate:VariableAggregate. What do you guys think?

indradeep commented on blurr/core/constants.py:
```diff
@@ -0,0 +1,9 @@
+BLURR_TRANSFORM_STREAMING = "Blurr:Transform:Streaming"
+BLURR_TRANSFORM_WINDOW = "Blurr:Transform:Window"
+BLURR_AGGREGATE_BLOCK = "Blurr:Aggregate:BlockAggregate"
```
Let us include the change we discussed here - for Aggregates, remove the superfluous `Aggregate` in the end.  Remember to do a mass replace form the documentation too!

vigbk replied:
Will do

prashastk commented on blurr/core/constants.py:
```diff
@@ -0,0 +1,9 @@
+BLURR_TRANSFORM_STREAMING = "Blurr:Transform:Streaming"
```
How about creating a type.py file which defines these as an enum:
```
class Type(Enum):
    BLURR_TRANSFORM_STREAMING = "Blurr:Transform:Streaming",
    BLURR_AGGREGATE_BLOCK = "Blurr:Aggregate:BlockAggregate",
```

Then passing around Types will be clear as an object as opposed to arbitrary strings. This class can also encapsulate other type related functionality that we may want to add in the future.

vigbk replied:
Sure. Probably Call it DTCType to be more specific.

prashastk replied:
file being in blurr/core makes it about the DTC. don't think we need to specify that (just like we don't call other classes / objects as DTC)

vigbk replied:
sure

vigbk commented:
Okay sure

On Sat, Apr 21, 2018, 3:28 AM Prashast Khandelwal <notifications@github.com>
wrote:

> *@prashastk* commented on this pull request.
> ------------------------------
>
> In blurr/core/constants.py
> <https://github.com/productml/blurr/pull/228#discussion_r183148264>:
>
> > @@ -0,0 +1,9 @@
> +BLURR_TRANSFORM_STREAMING = "Blurr:Transform:Streaming"
>
> file being in blurr/core makes it about the DTC. don't think we need to
> specify that (just like we don't call other classes / objects as DTC)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/productml/blurr/pull/228#discussion_r183148264>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AF5yQHhYB96LynA971olxpAsNatrJ6LXks5tqjbGgaJpZM4TdfUG>
> .
>

vigbk commented:
Updated PR after converting from string constants to enums.
Also removed redundant suffixes from string types eg: Blurr:Aggregate:VariableAggregate -> Blurr:Aggregate:Variable.

I will update the documentation in another commit for ease of review.
